core/logger/logger.go

```diff
@@ -173,6 +188,18 @@ func (l *zapLogger) ErrorIfClosing(c io.Closer, name string) {
 	}
 }
 
+func (l *zapLogger) Critical(args ...interface{}) {
+	l.sugaredWithCallerSkip(1).DPanic(args...)
```

Comment:
I'm not sure DPanic fits the semantics of what we're trying to do here:

https://github.com/uber-go/zap/blob/master/FAQ.md#whats-dpanic

Reply:
We are not actually using zap in development mode at all, so we are only re-purposing the existing level ordering without opting-in for development panics. But regardless, why doesn't this map to our use case? I thought this sounded like just what we wanted, particularly if we are going to start recovering from panics again:
> errors that are theoretically possible, but shouldn't actually happen, without crashing in production

Reply:
Because as proposed critical would also be used for serious errors that are *expected to happen*

Reply:
Can you elaborate on what _expected_ means in this context? Any type of error is expected under the right conditions, but I would have thought that for a properly configured and maintained node we would not expect any Critical errors to be logged. Do we anticipate this being used as a regular notification mechanism, rather than limiting only to truly exceptional cases?